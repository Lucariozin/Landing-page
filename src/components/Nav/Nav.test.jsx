import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Nav } from '.';
import NavMock from './mock';

describe('<Nav />', () => {
  it('should render <Nav /> correctly', () => {
    render(<Nav linksData={NavMock} />);

    const menuLinks = screen.getAllByRole('link');
    const logo = screen.getByRole('heading', { name: 'logo' });
    const navContainer = logo.parentElement.parentElement;

    expect(menuLinks).toHaveLength(8);

    expect(logo).toBeInTheDocument();

    expect(navContainer).toHaveStyleRule('background-color', 'var(--white)');
    expect(navContainer).toHaveStyleRule('display', 'none', {
      media: '(max-width: 800px)',
    });
  });

  it('should open Menu with click in <ToggleMenu />', () => {
    const { container } = render(<Nav linksData={NavMock} />);

    const toggleMenu = container.firstChild;
    const logo = screen.getByRole('heading', { name: 'logo' });
    const navContainer = logo.parentElement.parentElement;

    expect(toggleMenu).toHaveStyleRule('display', 'flex', {
      media: '(max-width: 800px)',
    });

    userEvent.click(toggleMenu);

    expect(navContainer).toHaveStyleRule('display', 'flex', {
      media: '(max-width: 800px)',
    });

    userEvent.click(toggleMenu);

    expect(navContainer).toHaveStyleRule('display', 'none', {
      media: '(max-width: 800px)',
    });
  });

  it('should to match snapshot', () => {
    const { container } = render(<Nav linksData={NavMock} />);
    expect(container).toMatchSnapshot();
  });
});
