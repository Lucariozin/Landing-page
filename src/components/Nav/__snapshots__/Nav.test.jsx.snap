// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should to match snapshot 1`] = `
.c6 {
  display: block;
  font-size: 1.8rem;
  color: var(--primary-color);
  padding: 1.5rem;
  position: relative;
}

.c6::after {
  content: '';
  position: absolute;
  bottom: 0.3rem;
  left: 50%;
  width: 0;
  background-color: var(--secondary-color);
  height: 0.2rem;
  -webkit-transition: 300ms all ease-in-out;
  transition: 300ms all ease-in-out;
}

.c6:hover::after {
  left: 25%;
  width: 50%;
}

.c4 {
  color: var(--primary-color);
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c2 {
  background-color: var(--white);
  width: 100%;
  border-bottom: 0.1rem solid var(--light-gray-color);
  position: fixed;
  z-index: 2;
}

.c3 {
  max-width: 120rem;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.8rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  z-index: 3;
  width: 4rem;
  height: 4rem;
  padding-bottom: 0.2rem;
  background: var(--primary-color);
  color: var(--white);
  font-size: 3.3rem;
  cursor: pointer;
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 2rem;
  right: 2rem;
}

.c0::after {
  content: '☰';
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 1px solid var(--primary-color);
  background: var(--white);
  cursor: pointer;
  position: fixed;
  bottom: 2rem;
  right: 2rem;
}

@media (max-width:800px) {
  .c2 {
    display: none;
    min-height: 100vh;
    height: 100%;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:800px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:800px) {
  .c5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 2rem;
  }
}

@media (max-width:800px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div>
  <div
    class="c0"
  />
  <a
    class="c1"
    href="#"
    target="_self"
  >
     
    <span
      class=""
    >
      ➤
    </span>
     
  </a>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h1
        class="c4"
      >
        <a
          alt="logo"
          class=""
          href="#"
        >
          logo
        </a>
      </h1>
      <div
        class="c5"
      >
        <a
          class="c6"
          href="#intro"
          target="_self"
        >
          intro
        </a>
        <a
          class="c6"
          href="#grid-one"
          target="_self"
        >
          grid-one
        </a>
        <a
          class="c6"
          href="#gallery"
          target="_self"
        >
          gallery
        </a>
        <a
          class="c6"
          href="#grid-two"
          target="_self"
        >
          grid-two
        </a>
        <a
          class="c6"
          href="#pricing"
          target="_self"
        >
          pricing
        </a>
        <a
          class="c6"
          href="#contact"
          target="_self"
        >
          contact
        </a>
      </div>
    </div>
  </div>
</div>
`;
